src/ussd_nurse.js

```diff
@@ -1,21 +1,113 @@
 go.app = function() {
     var vumigo = require("vumigo_v02");
+    var _ = require('lodash');
     var App = vumigo.App;
-    var EndState = vumigo.states.EndState;
+    var Choice = vumigo.states.Choice;
+    var ChoiceState = vumigo.states.ChoiceState;
+    var PaginatedChoiceState = vumigo.states.PaginatedChoiceState;
+    // var EndState = vumigo.states.EndState;
+    var JsonApi = vumigo.http.api.JsonApi;
+
+    var SeedJsboxUtils = require('seed-jsbox-utils');
+    var IdentityStore = SeedJsboxUtils.IdentityStore;
+
+    var utils = SeedJsboxUtils.utils;
 
     var GoNDOH = App.extend(function(self) {
-        App.call(self, "state_start");
+        App.call(self, "state_route");
         var $ = self.$;
 
+        // variables for services
+        var is;
+
         self.init = function() {
+            // initialising services
+            is = new IdentityStore(
+                new JsonApi(self.im, {}),
+                self.im.config.services.identity_store.token,
+                self.im.config.services.identity_store.url
+            );
+
+            var msisdn = utils.normalize_msisdn(self.im.user.addr, '27');
+            return is
+            .get_or_create_identity({"msisdn": msisdn})
+            .then(function(identity) {
+                if ((!_.isUndefined(identity.nc_working_on))
+                    && (identity.nc_working_on !== "")) {
+                    self.im.user.set_answer("user", identity);
+                    return is
+                    .get_or_create_identity({"msisdn": identity.nc_working_on})
+                    .then(function(working_on_identity) {
+                        self.im.user.set_answer("contact", working_on_identity);
+                    });
+                } else {
+                    self.im.user.set_answer("user", identity);
+                    self.im.user.set_answer("contact", identity);
+                }
+            });
+        };
+
+        // override normal state adding
+        self.add = function(name, creator) {
+            self.states.add(name, function(name, opts) {
+                // if (!interrupt || !self.timed_out(self.im))
+                    return creator(name, opts);
 
+                // interrupt = false;
+                // var timeout_opts = opts || {};
+                // timeout_opts.name = name;
+                // return self.states.create('st_timed_out', timeout_opts);
+            });
         };
 
-        self.states.add("state_start", function(name) {
-            return new EndState(name, {
-                text: $("Welcome to The Department of Health's ussd_nurse.js"),
+    // DELEGATOR START STATE
+
+        self.add('state_route', function(name) {
+            // reset working_on extra
+            self.im.user.set_answer("nc_working_on", "");
+
+            var user = self.im.user.answers.user;
+            if (user.nc_is_registered === 'true') {
```

Comment:
I think it would be better and more in line with what we've done for pmtct to look if the contact has an active nurseconnect subscription, rather than looking at the extra.